src/pages/ValidateLoginPage/index.tsx

```diff
@@ -13,16 +14,29 @@ function ValidateLoginPage({
     session,
 }: ValidateLoginPageProps<ValidateLoginPageOnyxNativeProps>) {
     useEffect(() => {
-        if (session?.authToken) {
-            // If already signed in, do not show the validate code if not on web,
-            // because we don't want to block the user with the interstitial page.
-            Navigation.goBack();
-        } else {
-            Session.signInWithValidateCodeAndNavigate(Number(accountID), validateCode);
-        }
+        // Wait till navigation becomes available
+        Navigation.isNavigationReady().then(() => {
+            if (session?.authToken) {
+                // If already signed in, do not show the validate code if not on web,
+                // because we don't want to block the user with the interstitial page.
+                Navigation.goBack();
+            } else {
+                Session.signInWithValidateCodeAndNavigate(Number(accountID), validateCode);
+            }
+        });
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    useEffect(() => {
+        if (session?.autoAuthState !== CONST.AUTO_AUTH_STATE.FAILED) {
+            return;
+        }
+        // Go back to initial route if validation fails
+        Navigation.isNavigationReady().then(() => {
+            Navigation.goBack();
+        });
+    }, [session?.autoAuthState]);
```

Comment:
Were you able to find the root cause? How come it used to work before?

Reply:
I think it is because of navigation refactor. User should be redirected back to sign-in page if code validation fails.

Reply:
I know it's from navigation refactor. But wanted to know exactly which code caused this.
This file was not changed recently